### PR TITLE
Small clipping bug

### PR DESCRIPTION
**Pull request by skygering**
Small clipping bug fix. 

Are we okay with a patch bump?

### CODE REVIEW

File: src/methods/clipping/difference.jl [hunk 1]
```diff
@@ -162,6 +162,7 @@ function _difference(
     local polys
     for (i, poly_b) in enumerate(GI.getpolygon(multipoly_b))
         polys = difference(i == 1 ? multipoly_a : GI.MultiPolygon(polys), poly_b; target, fix_multipoly)
+        isempty(polys) && break
```
Comment (asinghvi17) [inline src/methods/clipping/difference.jl:165]: Doesn't this fail if multiply A only intersects the 2nd polygon in multiply B?

(Multiply = multi poly, on mobile)

**Comment by asinghvi17** [comment]
Patch bump is fine

File: src/methods/clipping/difference.jl (hunk 1 above)
Comment (asinghvi17) [reply src/methods/clipping/difference.jl:165]: Specifically, considering the case where `multipoly_a` is really just `MultiPolygon(first(GI.getgeom(multipoly_b)))`.  In this case, polys will be empty at i=1, but that short circuiting will ignore the other polygons in poly_b, which are different from multipoly_a.

File: src/methods/clipping/difference.jl [hunk 2]
```diff
@@ -162,6 +162,7 @@ function _difference(
     local polys
     for (i, poly_b) in enumerate(GI.getpolygon(multipoly_b))
         polys = difference(i == 1 ? multipoly_a : GI.MultiPolygon(polys), poly_b; target, fix_multipoly)
```
Comment (rafaqz) [inline src/methods/clipping/difference.jl:164]: We should split out this conditional into an is/else, this logic is a bit hard to follow.

Comment (rafaqz) [reply src/methods/clipping/difference.jl:164]: Maybe with a few comments... Given the bug and anshuls comment

File: src/methods/clipping/difference.jl (hunk 1 above)
Comment (skygering) [reply src/methods/clipping/difference.jl:165]: That doesn't matter because we are taking the difference from multipolygon `a`. So once multipolygon `a` is completely covered by any pieces of multipolgon `b`, we have no piece of multipolygon `a` left to take the difference from with the remaining `b` pieces.

Comment (skygering) [reply src/methods/clipping/difference.jl:165]: This is a unique short circuit that applies to `difference`. I 100% agree this type of logic doesn't work with `intersection` or `union`.

File: src/methods/clipping/difference.jl (hunk 2 above)
Comment (skygering) [reply src/methods/clipping/difference.jl]: Updated!

File: src/methods/clipping/difference.jl (hunk 1 above)
Comment (asinghvi17) [reply src/methods/clipping/difference.jl:174]: Gotcha - that  makes sense.

**Review by asinghvi17 — APPROVED** [review]
